src/main/java/uk/gov/ida/verifyserviceprovider/configuration/HubEnvironment.java

```diff
@@ -29,20 +88,99 @@ public static HubEnvironment fromString(String name) {
             .findFirst()
             .orElseThrow(() -> new RuntimeException(
                 "Unrecognised Hub Environment: " + name + ". \n" +
-                "Valid values are: PRODUCTION, INTEGRATION, COMPLIANCE_TOOL"
+                "Valid values are: PRODUCTION, INTEGRATION, COMPLIANCE_TOOL, CUSTOM"
             ));
     }
 
-    HubEnvironment(URI ssoLocation, URI metadataUri) {
+    HubEnvironment(String ssoLocation,
+                   String metadataUri,
+                   String expectedEntityId,
+                   String metadataTrustStorePath,
+                   String hubTrustStorePath,
+                   String idpTrustStorePath,
+                   String trustStorePassword,
+                   boolean loadTruststoreFromResources) {
         this.ssoLocation = ssoLocation;
         this.metadataUri = metadataUri;
+        this.expectedEntityId = expectedEntityId;
+        this.metadataTrustStorePath = metadataTrustStorePath;
+        this.hubTrustStorePath = hubTrustStorePath;
+        this.idpTrustStorePath = idpTrustStorePath;
+        this.trustStorePassword = trustStorePassword;
+        this.loadTruststoreFromResources = loadTruststoreFromResources;
     }
 
     public URI getSsoLocation() {
-        return this.ssoLocation;
+        return URI.create(this.ssoLocation);
+    }
+
+    @Override
+    public KeyStore getTrustStore() {
+        return loadTruststore(metadataTrustStorePath);
+    }
+
+    @Override
+    public Optional<KeyStore> getHubTrustStore() {
+        return Optional.ofNullable(loadTruststore(hubTrustStorePath));
+    }
+
+    @Override
+    public Optional<KeyStore> getIdpTrustStore() {
+        return Optional.ofNullable(loadTruststore(idpTrustStorePath));
+    }
+
+    @Override
+    public URI getUri() {
+        return URI.create(metadataUri);
+    }
+
+    @Override
+    public Long getMinRefreshDelay() {
+        return 60000L;
+    }
+
+    @Override
+    public Long getMaxRefreshDelay() {
+        return 600000L;
+    }
+
+    @Override
+    public String getExpectedEntityId() {
+        return expectedEntityId;
+    }
+
+    @Override
+    public JerseyClientConfiguration getJerseyClientConfiguration() {
+        return new JerseyClientConfiguration();
+    }
+
+    @Override
+    public String getJerseyClientName() {
+        return HUB_JERSEY_CLIENT_NAME;
+    }
+
+    @Override
+    public String getHubFederationId() {
+        return "VERIFY-FEDERATION";
+    }
+
+    private KeyStore loadTruststore(String trustStore) {
+        return loadTruststoreFromResources ? loadTruststoreFromResources(trustStore) : loadTruststoreFromFile(trustStore);
+    }
+
+    private KeyStore loadTruststoreFromFile(String trustStore) {
+        try {
+            return new KeyStoreLoader().load(new FileInputStream(trustStore), trustStorePassword);
```

Comment:
Can we use `try-with-resources` statement to ensure that the FileInputStream instance is closed after loading the file?

Reply:
Ah, there is a load() method which uses `try-with-resources` statement. This will ensure that  FileInputStream instance is closed.